tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py

```diff
@@ -0,0 +1,289 @@
+        log.info(f"\n{out}")
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root2")
+        log.info(f"\n{out}")
+
+        if (
+            df_utilization_root1_initial["summary"]["total_kb_used"]
+            < df_utilization_root1_step_1["summary"]["total_kb_used"]
+        ):
+            log.info(
+                f"Utilization has been increased after filling data before: "
+                f"{df_utilization_root1_initial['summary']['total_kb_used']}"
+                f"\n {df_utilization_root1_step_1['summary']['total_kb_used']}"
+            )
+        else:
+            raise CommandFailed(
+                (
+                    f"Utilization has not increased after filling data before: "
+                    f"{df_utilization_root1_initial['summary']['total_kb_used']}"
+                    f"\n {df_utilization_root1_step_1['summary']['total_kb_used']}"
+                )
+            )
+
+        if (
+            df_utilization_root2_initial["summary"]["total_kb_used"]
+            >= df_utilization_root2_step_1["summary"]["total_kb_used"]
+        ):
+            raise CommandFailed(
+                (
+                    f"Utilization has increased after filling data in root2 : "
+                    f"{df_utilization_root2_initial['summary']['total_kb_used']}"
+                    f"\n {df_utilization_root2_step_1['summary']['total_kb_used']}"
+                )
+            )
+        log.info("Change crush rule to root2")
+        client1.exec_command(
+            sudo=True, cmd="cp crush.map_root1.txt crush.map_root2.txt"
+        )
+        client1.exec_command(
+            sudo=True,
+            cmd="sed -i 's/step take root1/step take root2/' crush.map_root2.txt",
+        )
+        client1.exec_command(
+            sudo=True, cmd="crushtool -c crush.map_root2.txt -o crush_root2.map.bin"
+        )
+        client1.exec_command(
+            sudo=True, cmd="ceph osd setcrushmap -i crush_root2.map.bin"
+        )
+
+        time.sleep(120)
+        df_utilization_root1_step_2, rc = client1.exec_command(
+            sudo=True, cmd="ceph osd df root1 -f json"
+        )
+        df_utilization_root1_step_2 = json.loads(df_utilization_root1_step_2)
+
+        df_utilization_root2_step_2, rc = client1.exec_command(
+            sudo=True, cmd="ceph osd df root2 -f json"
+        )
+        df_utilization_root2_step_2 = json.loads(df_utilization_root2_step_2)
+        log.info("Step 2 df values for root1 and root2")
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root1")
+        log.info(f"\n{out}")
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root2")
+        log.info(f"\n{out}")
+        if (
+            df_utilization_root1_step_1["summary"]["total_kb_used"]
+            < df_utilization_root1_step_2["summary"]["total_kb_used"]
+        ):
+            raise CommandFailed(
+                f"Data not migrated to the root2 root1 before :"
```

Comment:
"Data not migrated from root1 to root2" gives more clarity, also add \n for before and after outputs.